src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -27,9 +27,17 @@ import org.jetbrains.kotlinx.lincheck.strategy.*
 import org.jetbrains.kotlinx.lincheck.strategy.managed.*
 import org.jetbrains.kotlinx.lincheck.verifier.*
 import java.lang.reflect.*
-import kotlin.math.max
 import kotlin.random.*
 
+internal interface Interleaving {
```

Comment:
Do we need this interface when you almost always use inheritors of `AbstractSwitchesInterleaving`?
If not, lets get rid of it and simplify `AbstractSwitchesInterleaving` name to `SwitchestInterleaving`.
Similarly, `AbstractSwitchesInterleavingBuilder` -> `SwitchesInterleavingBuilder`.
Abstract `InterleavingTreeNode` does not have `Abstract`

Reply:
> Abstract InterleavingTreeNode does not have Abstract
I think InterleavingTreeNode is shorter and looks better. Having so many concrete inheritors InterleavingTreeNode looks like something common, so it is clear that it is abstract. No?

Reply:
I cannot get rid of ModelCheckingStrategy.SwitchesInterleaving as I have to specify a type parameter for super class. The same with builder - a concrete result of build can be specified only in inheritor. In this case, I think basic interfaces make understanding type parameters easier.

If you have some ideas how to make a simpler compilable version of this code, could we discuss it together?

Reply:
> I think InterleavingTreeNode is shorter and looks better.
That's my point. I just showed that not all abstract classes have `Abstract` and suggested to shorten these names

> I cannot get rid of ModelCheckingStrategy.SwitchesInterleaving
But I think you can get rid of `AbstractSwitchesInterleaving`

Reply:
Ok, I got it. How about revert names back to simple Interleaving and InterleavingBuilder?

Reply:
Yeah, sounds good